lib/krikri/search_results_helper_behavior.rb

```diff
@@ -42,5 +42,51 @@ def render_bookmarks_control?
       false
     end
 
+    # Render enriched record for view
+    # @param [Krikri::SearchIndexDocument]
+    # @return [String]
+    def render_enriched_record(document)
+      agg = document.aggregation
+      return error_msg unless agg.present?
+      JSON.pretty_generate(agg.to_jsonld)
+    end
+
+    # Render original record for view
+    # @param [Krikri::SearchIndexDocument]
+    # @return [String]
+    def render_original_record(document)
+      agg = document.aggregation
+      return error_msg unless agg.present?
+
+      begin
+        original_record = agg.original_record
+      rescue StandardError => e
+        logger.error e.message
+        return error_msg
+      end
+
+      prettify_string(original_record.to_s)
+    end
+
+    private
+
+    def prettify_string(string)
+      # return JSON if original record content is JSON
+      begin
+        return JSON.pretty_generate(JSON.parse(string))
+      rescue JSON::ParserError
+      end
+
+      # return XML String if original record content is XML
+      return Nokogiri::XML(string).to_xml if
+        Nokogiri::XML(string).errors.empty?
+
+      # return String
+      string
+    end
+
+    def error_msg
+      'There was a problem getting the record.'
```

Comment:
You're catching the exception in `#render_original_record`, and that calls `#error_msg`. Can `#render_original_record` also include the error message in what gets rendered?